pyquil/quilbase.py

```diff
@@ -199,16 +200,24 @@ def out(self) -> str:
                 _format_qubits_out(self.qubits),
             )
 
-    def controlled(self, control_qubit: QubitDesignator) -> "Gate":
+    def controlled(
+        self, control_qubit: Union[QubitDesignator, Sequence[QubitDesignator]]
+    ) -> "Gate":
         """
-        Add the CONTROLLED modifier to the gate with the given control qubit.
+        Add the CONTROLLED modifier to the gate with the given control qubit or Sequence of control
+        qubits.
         """
-        control_qubit = unpack_qubit(control_qubit)
-
-        self.modifiers.insert(0, "CONTROLLED")
-        self.qubits.insert(0, control_qubit)
-
-        return self
+        if isinstance(control_qubit, Sequence):
+            for qubit in control_qubit:
+                qubit = unpack_qubit(qubit)
+                self.modifiers.insert(0, "CONTROLLED")
+                self.qubits.insert(0, qubit)
+            return self
+        else:
+            control_qubit = unpack_qubit(control_qubit)
+            self.modifiers.insert(0, "CONTROLLED")
+            self.qubits.insert(0, control_qubit)
+            return self
```

Comment:
You could get rid of a bit of code duplication here (for a small performance cost) with something like:

```python
control_qubits = control_qubit if isinstance(control_qubit, Sequence) else [control_qubit]
for qubit in control_qubits:
    ...
```

Reply:
Done